lib/kennel/utils.rb

```diff
@@ -57,6 +57,11 @@ def color(color, text, force: false)
         "\e[#{COLORS.fetch(color)}m#{text}\e[0m"
       end
 
+      def truncate_lines(text, to:, warning:)
+        return text if text.count("\n") < to
+        text.split(/\n/).first(to).join("\n") + "\n" + warning
```

Comment:
More efficient at runtime, harder to read:

```suggestion
        text.split(/\n/, to + 1).first(to).join("\n") + "\n" + warning
```

More than happy to keep it as it is though.

Reply:
(Untested) Simpler, and also more consistent with newline handling maybe? Effectively requires that `text` and `warning` both end with a newline.

```suggestion
        lines = text.lines.take(to + 1)
        lines[-1] = warning if lines.length > to
        lines.join
```

Reply:
I like just having 1 path through the code 👍